bot/models.py

```diff
@@ -8,129 +8,186 @@
 from typing import Optional
 from typing import Optional, Any
 from bot.config import settings
-from bot.config import settings
 from dataclasses import dataclass
 import datetime
 import maya
 import json
 
-def safe_int(value: Any, fallback: int) -> int:
-    try:
-        return int(value)
-    except:
-        return fallback
+
+@dataclass
+class kappaItemsModel:
+    name: str
+    quantity: int
+
+    @classmethod
+    def fromJSONObj(cls, object: Any) -> kappaItemsModel:
+        return kappaItemsModel(
+            name=object.get("name"),
+            quantity=object.get("itemQTY"),
+        )
 
 
 @dataclass
-class TarkovMarketModel:
+class kappaQuestsModel:
     name: str
-    shortName: str
-    price: int
-    basePrice: int
-    avg24hPrice: int
-    avg7daysPrice: int
-    traderName: str
-    traderPrice: int
-    tracePriceCur: str
-    updated: datetime.datetime
-    slots: int
-    img: str
-    imgBig: str
+    isReq: int
 
     @classmethod
-    def fromJSONObj(cls, object: Any) -> TarkovMarketModel:
-        return TarkovMarketModel(
+    def fromJSONObj(cls, object: Any) -> kappaQuestsModel:
+        return kappaQuestsModel(
             name=object.get("name"),
-            shortName=object.get("shortName"),
-            price=safe_int(object.get("price"), 0),
-            basePrice=safe_int(object.get("basePrice"), 0),
-            avg24hPrice=safe_int(object.get("avg24hPrice"), 0),
-            avg7daysPrice=safe_int(object.get("avg7daysPrice"), 0),
-            traderName=object.get("traderName"),
-            traderPrice=safe_int(object.get("traderPrice"), 0),
-            tracePriceCur=object.get("tracePriceCur"),
-            updated=maya.parse(object.get("updated")).datetime(),
-            slots=safe_int(object.get("slots"), 0),
-            img=object.get("img"),
-            imgBig=object.get("imgBig"),
+            isReq=object.get("kappaReq"),
         )
 
+
 @dataclass
-class TarkovDatabaseModel:
+class LogicalHelmetModel:
+    bsgID: str
     name: str
-    shortName: str
+    armorClass: str
+    armorZones: str
+    armorDurability: str
+    armorRico: str
+    armorMoveSpeed: str
+    armorTurnSpeed: str
+    armorErgo: str
+    helmetSoundReduc: str
+    helmetBlocksHeadset: str
+    wikiLink: str
     description: str
-    useTime: int
-    effects: str
 
     @classmethod
-    def fromJSONObj(cls, object: Any) -> TarkovDatabaseModel:
-        return TarkovDatabaseModel(
+    def fromJSONObj(cls, object: Any) -> LogicalHelmetModel:
+        return LogicalHelmetModel(
+            bsgID=object.get("bsgID"),
```

Comment:
note that `object.get("somekey")` is a nullable expression. All of these properties should be non-null, according to the class you wrote above. Convert them all to `Optional[str]`, or make the accesses below not-nullable (`object["name"]`).

Reply:
Note that I Highly recommend that we don't make the model nullable everywhere, and instead

1. provide default values for numbers that aren't there (`safe_int`'s second parameter)
2. try: except: around parsing all our other keys, and use `dict["thiskey"]` instead of `dict.get("thiskey")`

Reply:
To be super super clear on what to do here;
- Leave all the classes as they are, without optional anywhere.
- Anywhere you're parsing out the JSON keys (calling object.get("key") today), use object["key"] and surround the outside with `try: except KeyError as e:`